scripts/static-analyser-check.sh

```diff
@@ -1,10 +1,12 @@
 #!/bin/bash
 
+set -exo pipefail
+
 # Ensure we are at the project root
 cd "$(dirname $0)"/..
 
 # Collect scan-build output
 ninja scan-build -C builddir | tee builddir/meson-logs/scan-build.txt
 
 # Invert the output - if this string exists it's a fail
-exit ! grep -E '[0-9]+ bugs? found.' builddir/meson-logs/scan-build.txt
+! grep -E '[0-9]+ bugs? found.' builddir/meson-logs/scan-build.txt
```

Comment:
`exit` only allows integers.  However, it doesn't make sense to use exit anyways since bash supports inverting exit codes.